File: fingertip/src/main/java/com/praszapps/fingertip/MVP/FingertipMVPContract.java
package com.praszapps.fingertip.MVP;

public interface FingertipMVPContract {

    interface View {
        void setUpFingerprintViews();
        void handleNoFingerprintAuthPossible();
        void onFingerPrintAuthenticationSuccess();
        void onFingerprintAuthenticationFailed(String text);

    }

    interface Presenter {
        void authenticateViaFingerprint();
        void stopFingerprintTracking();
    }

    interface Model {
        void getStandardErrorText();
    }

    interface IFingerprintRepository {
        void initalize();
        void startFingerprintTracking();
        void stopFingerprintTracking();
        boolean isFingerprintAvailable();
    }
}

File: fingertip/src/main/java/com/praszapps/fingertip/MVP/FingertipMVPContract.kt
package com.praszapps.fingertip.MVP

import android.app.KeyguardManager
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
import com.praszapps.fingertip.model.repository.ErrorModel
import com.praszapps.fingertip.presenter.FingertipDialogFragmentPresenter
import io.reactivex.Observable

internal interface FingertipMVPContract {

    interface IView {
        fun setUpFingerprintViews()
        fun onFingerPrintAuthenticationSuccess()
        fun onFingerprintAuthenticationFailed(text: String)

    }

    interface IPresenter {
        fun initialize(mFingerprintManager: FingerprintManagerCompat, mKeyguardManager: KeyguardManager)
        fun authenticateViaFingerprint()
        fun cancelFingerprintDetection()
    }

    interface IProvider {
        fun initialize(mFingerprintManager: FingerprintManagerCompat, mKeyguardManager: KeyguardManager): Observable<ErrorModel>
        fun startFingerprintTracking(callback: FingertipDialogFragmentPresenter.FingerprintResultCallback)
        fun stopFingerprintTracking()
    }
}
